### DISCUSSION TITLE
Why is MIDI note velocity map between 0 to 1 values?

### DISCUSSION THREAD

**Discussion by fpassa** [djipco/webmidi#157] [General]
I've just realized that velocity parameter of the note is map from 0 to 1 when all the MIDI industry uses 0 to 127. Why is this implemented in this manner?

May you please use the standards on velocity value range? I'm using 2.5.3 Webmidi version.

Thanks.
Flavio

**Comment by djipco** [comment]
The goal was to abstract the encoding from the intention. MIDI 1.0 uses 7 bits to encode velocity data. The new MIDI 2.0 uses 16 bits. By using a normalized value (0-1), I am trying to future-proof the library. Also, it's usually easier to deal with normalized values if the data is shared with a library, a third-party tool, etc.

However, if you prefer to use the 7 bit values, you can always specify `rawVelocity=false` in the options of the `playNote()` method.

By the way, there was [a bit of discussion](https://webmidijs.org/forum/discussion/comment/136#Comment_136) in the forum about this. I'm still open to ideas for v3 (which I'm currently working on).
